File: AR-VR/devAyushDubey_ayushdubey_2024csit1122_2/FunctionAndArrays/RotateArray.cpp
#include<iostream>
#include<cmath>
using namespace std;

void rotate(long int* arr,long int* arr2, int n,int k){
    int i,j=n;
    if(k<0)
        k+=n;
    for(i=k-1;i>=0;i--)
        arr2[i] = arr[--j];
    j=0;
    for(i=k;i<n;i++)
        arr2[i]=arr[j++];
   // write your code here
    
}

void display(long int* arr2, int n){
    for(int i = 0 ; i < n; i++){
        cout<<arr2[i]<<" ";
    }
    cout<<endl;
}

int main(){
    int n,k;
    cin>>n;
    long int arr[n],arr2[n],r;
    for(int i = 0 ; i < n; i++){
        cin>>arr[i];
    }
    cin>>r;
    k = r;
    if(r>n || r<(-1 * n))
        k = r % n;
    rotate(arr,arr2,n,k);
    display(arr2,n);
}

File: IOT/anshulgurjar53_AnshulSinghGurjar_2024it1193_2/FunctionAndArrays/rotatearray.java
import java.io.*;
import java.util.*;

public class Main{
  public static void rotate(int[] a, int k){
    // write your code here
    k=k%a.length;
    if(k>0){
    for(int i=k-1;i<a.length;i++)
    System.out.print(a[i]+" ");
    for(int i=0;i<k-1;i++)
    System.out.print(a[i]+" ");
    }
    if(k<0){
      k=k*-1;
      for(int i=k;i<a.length;i++)
      System.out.print(a[i]+" ");
      for(int i=0;i<k;i++)
      System.out.print(a[i]+" ");
    }
    if(k==0){
      for(int i=0;i<a.length;i++)
      System.out.println(a[i]+" ");
    }
  }

public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    int n = Integer.parseInt(br.readLine());
    int[] a = new int[n];
    for(int i = 0; i < n; i++){
       a[i] = Integer.parseInt(br.readLine());
    }
    int k = Integer.parseInt(br.readLine());

    rotate(a, k);
    
 }

}